apps/design/backend/src/db/db.ts

```diff
@@ -36,11 +42,26 @@ export class Db {
     });
   }
 
+  async close(): Promise<void> {
+    this.pool.removeAllListeners();
+    await this.pool.end();
+  }
+
   async withClient<T>(fn: (client: Client) => Promise<T>): Promise<T> {
     const poolClient = await this.pool.connect();
-    const client = new Client(await this.pool.connect());
+    const client = new Client(poolClient);
 
     try {
+      // Enable test suites to run concurrently on separate DB schemas.
+      // The default schema search path needs to be set on a per-connection
+      // basis.
+      // [TODO] Figure out if there's a better way to do this.
```

Comment:
What do you feel are the disadvantages to how this is currently done?

Reply:
Thought there might be an environment-wide flag we can set somewhere, instead of having to do this every time a client's pulled from the pool, but it's not looking likely - and this is only in the test path anyway, so not really a big deal, now that I think about it